pkg/ebpf/c/tracee.bpf.c

```diff
@@ -3592,6 +3594,176 @@ int BPF_KPROBE(trace_tcp_connect)
     return net_map_update_or_delete_sock(ctx, DEBUG_NET_TCP_CONNECT, sk, data.context.host_tid);
 }
 
+static __always_inline int icmp_delete_network_map(struct sk_buff *skb, int send, int ipv6) {
+
+    local_net_id_t connect_id = {0};
+    __u8 icmp_type;
+
+    if (ipv6) {
+        connect_id.protocol = IPPROTO_ICMPV6;
+
+        struct ipv6hdr *ip_header = (struct ipv6hdr *)(READ_KERN(skb->head) + READ_KERN(skb->network_header));
+        struct in6_addr daddr = READ_KERN(ip_header->daddr);
+        struct in6_addr saddr = READ_KERN(ip_header->saddr);
+
+        struct icmp6hdr *icmph = (struct icmp6hdr *)(READ_KERN(skb->head) + READ_KERN(skb->transport_header));
+        icmp_type = READ_KERN(icmph->icmp6_type);
+
+        connect_id.port = READ_KERN(icmph->icmp6_dataun.u_echo.identifier);
+
+        if (send) {
+            connect_id.address = daddr;
+        }
+        else {
+            if (icmp_type == ICMPV6_ECHO_REQUEST) {
+                return 0;
+            }
+
+            connect_id.address = saddr;
+        }
+    }
+    else {
+        connect_id.protocol = IPPROTO_ICMP;
+
+        struct iphdr *ip_header = (struct iphdr *)(READ_KERN(skb->head) + READ_KERN(skb->network_header));
+        __be32 daddr = READ_KERN(ip_header->daddr);
+        __be32 saddr = READ_KERN(ip_header->saddr);
+
+        struct icmphdr *icmph = (struct icmphdr *)(READ_KERN(skb->head) + READ_KERN(skb->transport_header));
+        icmp_type = READ_KERN(icmph->type);
+
+        connect_id.port = READ_KERN(icmph->un.echo.id);
+
+        if (send) {
+            connect_id.address.s6_addr32[3] = daddr;
+        }
+        else {
+            if (icmp_type == ICMP_ECHO) {
+                return 0;
+            }
+            #ifdef ICMP_EXT_ECHO
```

Comment:
We should remove this, right ? You're always defining it in `missing_definitions.h`, no ? And, considering this is an enum value defined in some kernels, and not others, the correct CO-RE way of using it would be to check for enum (like described at https://nakryiko.com/posts/bpf-core-reference-guide/#bpf-core-enum-value-exists).

Reply:
but the thing is that it is not an enum, its a define (https://elixir.bootlin.com/linux/v5.17.1/source/include/uapi/linux/icmp.h#L72).
if i understand correctly, `bpf_core_enum_value_exists()` doesn't help in this case. or am i missing something?

Reply:
No, you are not. I confused things when reviewing. I see it clearly now and it makes sense to always define it in CO-RE and to check type. Sorry for the noise.